x-pack/plugins/security_solution/public/entity_analytics/components/risk_score_preview_section.tsx

```diff
@@ -0,0 +1,191 @@
+  EuiFormRow,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiCallOut,
+  EuiButton,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+import type { BoolQuery, Query } from '@kbn/es-query';
+import { buildEsQuery } from '@kbn/es-query';
+import { RiskScoreEntity } from '../../../common/risk_engine/types';
+import { RiskScorePreviewTable } from './risk_score_preview_table';
+import * as i18n from '../translations';
+import type { RiskScore } from '../../../server/lib/risk_engine/types';
+import { useRiskScorePreview } from '../api/hooks/use_preview_risk_scores';
+import { useKibana } from '../../common/lib/kibana';
+import { SourcererScopeName } from '../../common/store/sourcerer/model';
+import { useSourcererDataView } from '../../common/containers/sourcerer';
+
+interface IRiskScorePreviewPanel {
+  showMessage: string;
+  hideMessage: string;
+  isLoading: boolean;
+  items: RiskScore[];
+  type: RiskScoreEntity;
+}
+
+const getRiskiestScores = (scores: RiskScore[] = [], field: string) =>
+  scores
+    ?.filter((item) => item?.identifierField === field)
+    ?.sort((a, b) => b?.totalScoreNormalized - a?.totalScoreNormalized)
+    ?.slice(0, 5) || [];
+
+const RiskScorePreviewPanel = ({
+  items,
+  showMessage,
+  hideMessage,
+  isLoading,
+  type,
+}: IRiskScorePreviewPanel) => {
+  const [trigger, setTrigger] = useState<'closed' | 'open'>('open');
+  const onToggle = (isOpen: boolean) => {
+    const newState = isOpen ? 'open' : 'closed';
+    setTrigger(newState);
+  };
+
+  return (
+    <EuiPanel hasBorder={true}>
+      <EuiAccordion
+        initialIsOpen={true}
+        isLoading={isLoading}
+        id={'host-table'}
+        buttonContent={trigger === 'closed' ? showMessage : hideMessage}
+        forceState={trigger}
+        onToggle={onToggle}
+        extraAction={<EuiIcon type={type === RiskScoreEntity.host ? 'storage' : 'user'} />}
+      >
+        <>
+          <EuiSpacer size={'m'} />
+          <RiskScorePreviewTable items={items} type={type} />
+        </>
+      </EuiAccordion>
+    </EuiPanel>
+  );
+};
+
+export const RiskScorePreviewSection = () => {
+  const [query, setQuery] = useState<{
+    dateRange: { from: string; to: string };
+    query: Query;
+  }>({
+    dateRange: { from: 'now-24h', to: 'now' },
+    query: { query: '', language: 'kuery' },
+  });
+
+  const {
+    unifiedSearch: {
+      ui: { SearchBar },
+    },
+  } = useKibana().services;
+
+  let riskScorePreviewFilter: { bool: BoolQuery } = {
+    bool: { must: [], filter: [], should: [], must_not: [] },
+  };
+  try {
+    riskScorePreviewFilter = buildEsQuery(undefined, query.query, []);
+  } catch (e) {
+    // Default to keeping default filter
+  }
+
+  const { data, isLoading, refetch, isError } = useRiskScorePreview({
+    filter: riskScorePreviewFilter,
+    range: {
+      start: query.dateRange.from,
+      end: query.dateRange.to,
+    },
+  });
+
+  const { indexPattern } = useSourcererDataView(SourcererScopeName.detections);
```

Comment:
`indexPattern` here appears deprecated. We should discuss with @dhurley14 whether this is an acceptable usage, or whether there's an alternative we should be using.

Reply:
Hey, @dhurley14 I tried to use `sourcererDataView`, but I can cast them to DataView[] types, as I can do with index patterns

```    <SearchBar
            appName="siem"
            isLoading={isLoading}
            indexPatterns={[indexPattern] as DataView[]}
            // query={query.query}
 ```
 
 indexPatterns is deprecated, but I can't find a better way to pass indexPatterns in SearchBar
 
 Here the types of SearchBar
 
` src/plugins/unified_search/public/search_bar/search_bar.tsx`

Reply:
You can do something similar to what @kqualters-elastic did here:

https://github.com/elastic/kibana/blob/4346b09760e962dd968fc2472bec65f476483dd1/x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/insight/index.tsx#L286-L294

Reply:
Spoke with Nikita: It looks like the data view spec is ignoring the narrowed index patterns and only using the data view id, which in this case is the default security view which. Doesn't seem to be behaving correctly so feel free to use the `indexPattern` here for now.